src/situations/commun/infra/depot_ressources.js

```diff
@@ -0,0 +1,63 @@
+function chargeurAudio (src) {
+  const audio = new window.Audio(src);
+
+  return new Promise((resolve, reject) => {
+    audio.addEventListener('canplaythrough', () => {
+      resolve(() => new window.Audio(src));
+    });
+
+    audio.addEventListener('error', reject);
+  });
+}
+
+function chargeurImage (src) {
+  const img = new window.Image();
+  const promesse = new Promise((resolve, reject) => {
+    img.onload = () => {
+      resolve(() => {
+        const clone = new window.Image();
+        clone.src = src;
+        return clone;
+      });
+    };
+
+    img.onerror = reject;
+  });
+  img.src = src;
+  return promesse;
+}
+
+const CHARGEURS = {
+  'mp3': chargeurAudio,
+  'png': chargeurImage,
+  'svg': chargeurImage
+};
+
+export default class DepotRessources {
+  constructor (chargeurs = CHARGEURS) {
+    this.chargeurs = chargeurs;
+    this.promesses = [];
+    this.ressources = {};
```

Comment:
Ça m'a pris un certain temps pour comprendre que `this.ressources` était une collection de "getters" de ressources. Je me demande si on peut améliorer le nommage de cette collection pour faciliter cette compréhension. Peut-être `this.accesseursRessource` ?.